src/corehost/cli/deps_resolver.cpp

```diff
@@ -403,6 +417,69 @@ void deps_resolver_t::init_known_entry_path(const deps_entry_t& entry, const pal
     }
 }
 
+void deps_resolver_t::resolve_additional_deps(const hostpolicy_init_t& init)
+{
+    pal::string_t additional_deps_serialized = init.additional_deps_serialized;
+    pal::string_t fx_name = init.fx_name;
+    pal::string_t fx_ver = init.fx_ver;
+
+    if (additional_deps_serialized.empty())
+    {
+        return;
+    }
+
+    pal::string_t additional_deps_path;
+    pal::stringstream_t ss(additional_deps_serialized);
+
+    // Process the delimiter separated custom deps files.
+    while (std::getline(ss, additional_deps_path, PATH_SEPARATOR))
+    {
+        // If it's a single deps file, insert it in 'm_additional_deps_files'
+        if (ends_with(additional_deps_path, _X(".deps.json"), false))
+        {
+            if (pal::file_exists(additional_deps_path))
+            {
+                m_additional_deps_files.push_back(additional_deps_path);
+            }
+        }
+        // If it's not a single deps file, we'll assume it's a directory
+        else
+        {
+            // We'll search deps files in 'base_dir'/shared/fx_name/fx_ver
+            append_path(&additional_deps_path, _X("shared"));
+            append_path(&additional_deps_path, fx_name.c_str());
```

Comment:
We could do something like that, but it wouldn't work in all scenarios.  Since the app is standalone, the user could have:

1. Not referenced M.N.App in 1.x.  (I know this isn't a mainline scenario, but it is possible.)
2. Use a pre-release version of M.N.App that the host doesn't have installed, thus doesn't have "light up".deps.json files for it.  For example, someone could take a 2.0.0-preview1 standalone app and publish to Azure today.
3. Use their own build-from-source version of M.N.App, which again the host would have no idea about.